EIPS/eip-140.md

```diff
@@ -0,0 +1,50 @@
+## Preamble
+
+    EIP: 140
+    Title: REVERT instruction in the Ethereum Virtual Machine
+    Author: Alex Beregszaszi, Nikolai Mushegian (nikolai@nexusdev.us)
+    Type: Standard Track
+    Category: Core
+    Status: Draft
+    Created: 2017-02-06
+
+## Simple Summary
+
+The `REVERT` instruction provides a way to stop execution and revert state changes, without consuming all provided gas and with the ability to return a reason.
+
+## Abstract
+
+The `REVERT` instruction will stop execution, roll back all state changes done so far and provide a pointer to a memory section, which can be interpreted as an error code or message. While doing so, it will not consume all the remaining gas.
+
+## Motivation
+
+Currently this is not possible. There are two practical ways to revert a transaction from within a contract: running out of gas or executing an invalid instruction. Both of these options will consume all remaining gas. Additionally, reverting a transaction means that all changes, including LOGs, are lost and there is no way to convey a reason for aborting a transaction.
+
+## Specification
+
+The `REVERT` instruction is introduced at `0xfd`. Execution is aborted, considered as failed, and state changes are rolled back.
+
+It expects two stack items, the top item is the `memory_offset` followed by `memory_length`. Both of these can equal to zero. The cost of the `REVERT` instruction equals to that of the `RETURN` instruction.
```

Comment:
I think it's worth specifying how the output memory contents are accessible.  After a `REVERT` during a message call, the output memory contents are accessible exactly the same way as if `RETURN` was executed.  After a `REVERT` during a contract creation, however, the output memory contents are not going to be deployed (unlike after a `RETURN` during a contract creation).  In both cases, the output memory contents are accessible from `RETURNDATASIZE` and `RETURNDATACOPY`, unless the `REVERT` was executed in the outermost context of a transaction.